consensus/round/round.go

```diff
@@ -93,12 +93,21 @@ func (rnd *round) UpdateRound(genesisTimeStamp time.Time, currentTimeStamp time.
 	rnd.updateRound(baseTimeStamp, currentTimeStamp, startRound, roundDuration)
 }
 
+// this should be called under mutex protection
 func (rnd *round) isSupernovaRoundActivated() bool {
-	return rnd.enableRoundsHandler.IsFlagEnabledInRound(common.SupernovaRoundFlag, uint64(rnd.index))
+	index := rnd.index
```

Comment:
could you move the read lock around the rnd.index read instead of on the caller?
then no need for the comment